src/Store/Features.re

```diff
@@ -186,6 +186,25 @@ let update =
     | None => (state, Effect.none)
     }
 
+  | Changelog(msg) =>
+    let eff =
+      switch (msg) {
+      | PullRequestClicked(pr) =>
+        Isolinear.Effect.create(~name="feature.changelog.openPR", () => {
+          let url =
+            Printf.sprintf("https://github.com/onivim/oni2/pull/%d", pr);
+          Revery.Native.Shell.openURL(url) |> ignore;
+        })
+
+      | CommitHashClicked(hash) =>
+        Isolinear.Effect.create(~name="feature.changelog.openCommit", () => {
+          let url =
+            Printf.sprintf("https://github.com/onivim/oni2/commit/%s", hash);
+          Revery.Native.Shell.openURL(url) |> ignore;
+        })
+      };
+    (state, eff);
```

Comment:
These really shouldn't live here, but in an update function in `Feature_Changelog` instead. And the effect ought to be defined in some library so it can be reused.

Reply:
Thanks for the feedback! I can move it to an update function. The reason I didn't do that was because I thought update functions were strictly for getting new state, which this doesn't do, but I guess I was wrong about that. 🤷‍♂️ 

I thought about adding a `OpenURLExternally` `Action`, but from my discussions with you and @bryphe I thought that sounded too imperative/"active". Is this something you think we would want?

Reply:
> I thought update functions were strictly for getting new state

Ah, yeah not really. It's for processing `msg`s. But it's very rare you'd have a feature window any state, so it's certainly not a bad approximation.

> I thought about adding a OpenURLExternally Action, but from my discussions with you and @bryphe I thought that sounded too imperative/"active". Is this something you think we would want?

You don't need an action, but your intuition is indeed also correct. That would be very imperative, which is what effects are for:

```
let openUri = uri =>
        Isolinear.Effect.create(~name="openUri", () =>
          Revery.Native.Shell.openURL(uri) |> ignore
        )
```

Reply:
Also, you should never `use `ignore` without specifying the type you ignore. That's just as bad as `let _ = ...`. It's functionally equivalent.

A nice way to use ignore with a type is:

```
ignore(Revery.Native.Shell.openURL(uri): whatever);
```

Reply:
imperative APIs that return somrthing other than `unit` is also not great. In fact a nice feature of APIs that return `unit` or take =unit` as its only argument is that it makes it obvious that it produces side-effects, as if it didn't it would be pointless.